LEAF_Request_Portal/templates/subindicators.tpl

```diff
@@ -353,6 +354,30 @@
                     }
                 });
             });
+            formValidator["id<!--{$indicator.indicatorID}-->"] = {
+                setValidator: function() {
+                    let regex = new RegExp(/^(?:(?:(?:0?[13578]|1[02])(\/|-|\.)31)\1|(?:(?:0?[1,3-9]|1[0-2])(\/|-|\.)(?:29|30)\2))(?:(?:1[6-9]|[2-9]\d)?\d{2})$|^(?:0?2(\/|-|\.)29\3(?:(?:(?:1[6-9]|[2-9]\d)?(?:0[48]|[2468][048]|[13579][26])|(?:(?:16|[2468][048]|[3579][26])00))))$|^(?:(?:0?[1-9])|(?:1[0-2]))(\/|-|\.)(?:0?[1-9]|1\d|2[0-8])\4(?:(?:1[6-9]|[2-9]\d)?\d{2})$/gm);
+                    return (regex.test($('#<!--{$indicator.indicatorID|strip_tags}-->').val()) || $('#<!--{$indicator.indicatorID|strip_tags}-->').val() == '');
+                },
+                setSubmitValid: function() {
+                    $([document.documentElement, document.body]).animate({
+                        scrollTop: $('#<!--{$indicator.indicatorID|strip_tags}-->_error').offset().top-50
+                    }, 700).clearQueue();
+                },
+                setValidatorError: function() {
+                    $('#<!--{$indicator.indicatorID|strip_tags}-->').css('border', '2px solid red');
```

Comment:
Just a comment.  Well.  Help me remember.  I think the team should have a discussion on what css we want to pull out of the "code" and what should be left in.  I'm of the mind things like this should be left in because while it matches other uses "2px solid red", in many ways it's a one-off for this specific class/function.

Reply:
Theres alot of CSS additions in all the tpls that can be attributed to classes and styled properly which would make everything so much more clean.

Reply:
it would also be easier to just add/remove classes than to have to modify the CSS in each call.

Reply:
Yeah thats what Carrie and I have been doing going forward after the debacle with that validation/error css we just made it a class.